core/components/scripts/fragment.ts

```diff
@@ -0,0 +1,26 @@
+import { graphql } from '~/client/graphql';
+
+export const ScriptsFragment = graphql(`
+  fragment ScriptsFragment on Content {
+    scripts(first: 50, filters: { visibilities: [ALL_PAGES, STOREFRONT] }) {
+      edges {
+        node {
+          __typename
+          integrityHashes {
+            hash
+          }
+          entityId
+          consentCategory
+          location
+          visibility
+          ... on InlineScript {
+            scriptTag
+          }
+          ... on SrcScript {
+            src
+          }
+        }
+      }
+    }
+  }
+`);
```

Comment:
We probably should actually be doing the filtering on the server rather than in the app:
```suggestion
export const ScriptsFragment = graphql(`
  fragment ScriptsFragment on Content {
    headScripts: scripts(first: 50, filters: { visibilities: [ALL_PAGES, STOREFRONT], location: [HEAD] }) {
      edges {
        node {
          ...ScriptFragment
        }
      }
    }
    footerScripts: scripts(first: 50, filters: { visibilities: [ALL_PAGES, STOREFRONT], location: [FOOTER] }) {
      edges {
        node {
          ...ScriptFragment
        }
      }
    }
  }
`, [ScriptFragment]);

const ScriptFragment = graphql(`
  __typename
  integrityHashes {
    hash
  }
  entityId
  consentCategory
  location
  visibility
  ... on InlineScript {
    scriptTag
  }
  ... on SrcScript {
    src
  }
`)
```

Reply:
Here I am pre-filtering for just any scripts relevant to Catalyst; scripts that render on the checkout page or the order confirmation page are irrelevant to Catalyst because it does not render those pages, so doing this filter at the API layer makes sense.

Reply:
>on the server

I'm referencing the BC servers/API layer here since I added `location: [FOOTER]` or `location: [HEAD]` as part of this suggestion